### ISSUE TITLE
Slow Training/VRAM Calculator

### ISSUE THREAD

**Issue by cgallik** [graphdeco-inria/gaussian-splatting#116]
Hello
TL/DR is there a way to calculate the amount of GPU VRAM required for training based on number of images and resolution?

I have been running some tests and trained a scene with 365 images at 3840x2160 reolution. initially they were downsampled at 1600x900 and it ran smoothly, albeit not the best output. I then rotated and reran at 1600x2844. This one is getting stuck at ~5800 iterations. I notice when it was loaded onto my GPU the total VRAM was about 22.1/24 GB, and estimated training time about an hour, but then it started spilling over to my "shared GPU memory" while it was training, and slows down significantly (~20 hours). 

I know I can specify loading it onto the CPU, but I would prefer knowing before training starts how to know if this would be necessary, and also to initially input the proper number of images when running the SfM pipeline so I might not have to use CPU.

**Comment by grgkopanas** [comment]
The cpu flag is not supposed to be so much slower. If in doubt use the cpu
flag.

Best,
George

**Comment by Snosixtyboo** [comment]
Hi,

yes, when Pytorch starts swapping between GPU and CPU, things get hellishly slow, unfortunately. It's hard to predict the memory that will be required ahead of time: if you have a region that has a lot of detail, but SfM failed there, the densification may have to put 100-500x more points than what it started with. If the SfM initialization reflects the detail in the scene well, then this multiplier can be much lower...

The CPU flag is a very nice intermediate solution. An even better solution would be to ditch PyTorch, which uses a lot of overhead memory, but we're not there yet.
